### ISSUE TITLE
Game flow: if user interacts, "Try Again" not required

### ISSUE THREAD

**Issue by ariel-phet** [phetsims/graphing-lines#47]
We have found in most games that it is a more seamless user experience that if a user gets a question wrong, and then goes to interact with the game it should have the same affect as pressing the "Try Again" button. This paradigm shift happened after the graphing lines game was developed, and it would be good to institute when we redeploy.

**Comment by pixelzoom** [comment]
Not clear from the description how this is supposed to work. Please provide a concrete example.

**Comment by ariel-phet** [comment]
See Area Builder....if you press the "check" button, and then the "try again" button is displayed, you can press the "try again" button OR you can just start manipulating pieces, either action produces the same result (the "try again" button goes to a "check" button)

**Comment by pixelzoom** [comment]
@ariel-phet Should a similar issue be created for balancing-chemical-equations and reactants-products-and-leftovers?

**Comment by pixelzoom** [comment]
@jbphet Can you please comment on how you've implemented this in Area Builder? Or point me to the relevant code? (Guessing you're using some sort of full-screen transparent overlay).

**Comment by pixelzoom** [comment]
Ah... it's not an overlay in Area Builder, because the button changes only when move a piece or click on the calculator.

**Comment by ariel-phet** [comment]
@pixelzoom - yes please create similar issues for BCE and RPAL...but they can be open questions for the design team. Graphing lines is a definite, but we should discuss those other two sims with the chem team.

**Comment by pixelzoom** [comment]
Hmm... This approach could work in RPAL if the spinners remain active.  But it's not going to work at all in BCE without significant redesign.  I'll create issues where we can discuss.

**Comment by pixelzoom** [comment]
Unassigning. No plans to work on this until this sim is revisited for publication.

**Comment by pixelzoom** [comment]
@ariel-phet Do you want me proceed with implementing this? Or is there some design work that needs to happen first?

**Comment by ariel-phet** [comment]
@pixelzoom I don't believe any design work is necessary, do you perceive any need for design work?

**Comment by pixelzoom** [comment]
I don't believe any design work is necessary, so I'll proceed.

**Comment by ariel-phet** [comment]
:boom:

**Comment by pixelzoom** [comment]
OK... Do I do have a design question. In Area Builder, you have to interact with specific things to make the "Try Again" button disappear. For example, see screenshot below from Level 1. Moving any blue square makes "Try Again" disappear. Toggling check boxes or clicking on the eraser button does not make "Try Again" disappear.  

So 2 questions:

(1) Is the current behavior of Area Build as desired, or is it buggy? (Clicking the eraser sure seems buggy to me.)

(2) Which interactions in Graphings Lines should remove the "Try Again" button? There are 3 things that can be interacted with in game challenges: spinners, points on the graph, and "point tools". Does interacting with one or more of these make the "Try Again" button disappear? Does clicking anywhere make it disappear?

**Comment by ariel-phet** [comment]
Assigning to @amanda-phet since she designed area builder and may have some good insights as to how to incorporate this game flow into graphing lines.

**Comment by pixelzoom** [comment]
I've noted the issue with the eraser button in Area Builder in https://github.com/phetsims/area-builder/issues/80.

**Comment by pixelzoom** [comment]
I discussed the Area Builder game with @jbphet.  The intent was to make the "Try Again" button disappear when the user did anything to change their answer.  @amanda-phet is that also appropriate for this sim?  If so, then changing spinners or moving points on the graph will make the "Try Again" button disappear, but moving point tools will not.

**Comment by amanda-phet** [comment]
If the user interacts with the graph or spinner, the "try again" button disappears and the "check" button will be in its place. I agree that moving the point tools does not have this effect since it would not change the answer in any way.

**Comment by pixelzoom** [comment]
Thanks @amanda-phet, I'll proceed with implementing this.

**Comment by pixelzoom** [comment]
This was a little more complicated than https://github.com/phetsims/reactants-products-and-leftovers/issues/37, because if there are >2 points, then the user's guess may not be a valid line. This was previously indicated by setting the 'guess' property to null - but this didn't provide a way to detect that the user's guess had changed.  So I replaced null with type NotALine, so that we have a new instance each time the user moves points, and can therefore detect that the guess has changed.

Demonstrated in this dev version:

http://www.colorado.edu/physics/phet/dev/html/graphing-lines/1.1.0-dev.2/graphing-lines_en.html

The "Place 3 Points" challenges are in Level 5. And you might find it more convenient to run with '?dev' so that you can skip challenges.

Assigning to @amanda-phet to verify.

**Comment by amanda-phet** [comment]
After getting this type of challenge wrong two times, I think feedback that "your line... is not a line!" is pretty valuable. When I get it wrong two times, and make sure the 3 points are not a line, it says my line is "y = x" which is not correct. 

So, if we change the properties as you say above, we still need to give some feedback at the end if the 3 points do not make a line. If they do make a line, the sim is already clear because there is a black line and it gives the equation in the feedback.

**Comment by pixelzoom** [comment]
@amanda-phet 
It sounds like you are reporting a new problem introduced by changing the game behavior. I will address that in a new issue in https://github.com/phetsims/graphing-lines/issues/49.

How about this issue? Is the game flow now as desired?

**Comment by amanda-phet** [comment]
Ah, I thought you wanted me to specifically look at the placing points challenges. The flow is definitely feeling good to me for all types of challenges. The placing points challenges are working as expected- if I change a point on the graph while the try again button is visible, the sim responds appropriately by letting me try again.

**Comment by pixelzoom** [comment]
OK, great. Then I'm going to close this issue.  The issue with "Place The Points" is in #49, and has also been resolved.
